js/app.js

```diff
@@ -135,42 +192,107 @@ function handleUpdateRateForm(event) {
     .value.trim()
     .toUpperCase();
 
-  // Get the value of the new exchange rate from the input field and convert it to a float
+  // Get the value of the new exchange rate from the input field
   const newExchangeRate = parseFloat(
     document.getElementById("update-exchange-rate").value
   );
 
-  // Find the rate object for the base currency
-  const rateObject = currencyRates.find((rate) => rate.base === baseCurrency);
+  // Find the rate
+  const rate = findCurrencyRate(baseCurrency, targetCurrency, false);
+
+  // If rate not found, show an error message
+  if (!rate) {
+    displayMessage(
+      `Exchange rate from ${baseCurrency} to ${targetCurrency} not found.`,
+      false
+    );
+    return;
+  }
+
+  // Update the rate
+  rate.exchangeRate = newExchangeRate;
 
-  if (rateObject && rateObject.rates[targetCurrency]) {
-    // If the rate object and the exchange rate for the target currency are found
-    rateObject.rates[targetCurrency] = newExchangeRate; // Update the exchange rate
+  // Reset the form fields
+  event.target.reset();
 
-    // Log the updated exchange rates array to the console
-    console.log("Updated exchange rates:", currencyRates);
+  // Update the displayed rates
+  renderCurrencyGrid();
+
+  // Show success message
+  displayMessage(
+    `Exchange rate from ${baseCurrency} to ${targetCurrency} updated successfully!`
+  );
+  console.log(
+    `Exchange rate from ${baseCurrency} to ${targetCurrency} updated successfully!`
+  );
+}
+
+// Function to handle search form
+function handleSearchForm(event) {
+  event.preventDefault();
+
+  const fromCurrency = document
+    .getElementById("search-from")
+    .value.trim()
+    .toUpperCase();
+  const toCurrency = document
+    .getElementById("search-to")
+    .value.trim()
+    .toUpperCase();
 
-    // Display success message
-    displayMessage("Exchange rate successfully updated!", true);
+  // Find the rate using the new function
+  let ratesList = [];
+  if (fromCurrency && toCurrency) {
+    ratesList.push(findCurrencyRate(fromCurrency, toCurrency, true));
+  } else if (fromCurrency && !toCurrency) {
+    ratesList = [...currencyRates].filter(
```

Comment:
`.filter()` returns a shallow copy already so you don't need to duplicate the array.